tools/read_cmt.ml

```diff
@@ -97,6 +97,64 @@ let print_info cmt =
   end;
   ()
 
+let generate_ml target_filename filename cmt =
+  let (printer, ext) =
+    match cmt.Cmt_format.cmt_annots with
+      | Cmt_format.Implementation typedtree ->
+          (fun ppf -> Pprintast.structure ppf
+                                        (Untypeast.untype_structure typedtree)),
+          ".ml"
+      | Cmt_format.Interface typedtree ->
+          (fun ppf -> Pprintast.signature ppf
+                                        (Untypeast.untype_signature typedtree)),
+          ".mli"
+      | _ ->
+        Printf.fprintf stderr "File was generated with an error\n%!";
```

Comment:
This error message may be good enough for a quick-and-dirty tool like `cmt2annot` but not for the compiler. You should probably raise an exception, and make sure the compiler raises `Fatal_error` in this case.

Reply:
This error message is only used for the tool `tools/cmt2annot` which continues to exist.